test/parallel/test-process-load-env-file.js

```diff
@@ -87,3 +87,57 @@ describe('process.loadEnvFile()', () => {
     assert.strictEqual(child.code, 0);
   });
 });
+
+describe('process.loadEnvFile(path, options)', () => {
+
+  it('should not override the original value', async () => {
+    process.env.BASIC = 'Original value';
+    const code = `
+      process.loadEnvFile(${JSON.stringify(validEnvFilePath)});
+      const assert = require('assert');
+      assert.strictEqual(process.env.BASIC, 'Original value');
+    `.trim();
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [ '--eval', code ],
+      { cwd: __dirname, env: process.env },
+    );
+    assert.strictEqual(child.stderr, '');
+    assert.strictEqual(child.code, 0);
+  });
+
+  it('should override the original value', async () => {
+    process.env.BASIC = 'Original value';
+    const code = `
+      const assert = require('assert');
+      assert.strictEqual(process.env.BASIC, 'Original value');
+      process.loadEnvFile(${JSON.stringify(validEnvFilePath)}, {override: true});
+      assert.strictEqual(process.env.BASIC, 'basic');
+    `.trim();
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [ '--eval', code ],
+      { cwd: __dirname,  env: process.env },
+    );
+    assert.strictEqual(child.stderr, '');
+    assert.strictEqual(child.code, 0);
+  });
+
+  it('supports passing options only', async () => {
+    process.env.BASIC = 'Original value';
```

Comment:
If you pass this variable in here, you mutate the current process which gets inherited by the spawn process call. But if you pass { env } as an option, it would not mutate the current process env vars.